xbmc/guilib/TextureFormats.h

```diff
@@ -25,4 +25,158 @@ enum XB_FMT
+  KD_TEX_FMT_ASTC_LDR_5x4       = 0x9100,
+  KD_TEX_FMT_ASTC_LDR_5x5       = 0x9200,
+  KD_TEX_FMT_ASTC_LDR_6x5       = 0x9300,
+  KD_TEX_FMT_ASTC_LDR_6x6       = 0x9400,
+  KD_TEX_FMT_ASTC_LDR_8x5       = 0x9500,
+  KD_TEX_FMT_ASTC_LDR_8x6       = 0x9600,
+  KD_TEX_FMT_ASTC_LDR_8x8       = 0x9700,
+  KD_TEX_FMT_ASTC_LDR_10x5      = 0x9800,
+  KD_TEX_FMT_ASTC_LDR_10x6      = 0x9900,
+  KD_TEX_FMT_ASTC_LDR_10x8      = 0x9A00,
+  KD_TEX_FMT_ASTC_LDR_10x10     = 0x9B00,
+  KD_TEX_FMT_ASTC_LDR_12x10     = 0x9C00,
+  KD_TEX_FMT_ASTC_LDR_12x12     = 0x9D00,
+
+  // ASTC HDR texture family
+  // Bitrate varies from 8bpp (4x4 tile) to 0.89bpp (12x12 tile).
+  KD_TEX_FMT_ASTC_HDR           = 0xA000,
+  KD_TEX_FMT_ASTC_HDR_4x4       = 0xA000,
+  KD_TEX_FMT_ASTC_HDR_5x4       = 0xA100,
+  KD_TEX_FMT_ASTC_HDR_5x5       = 0xA200,
+  KD_TEX_FMT_ASTC_HDR_6x5       = 0xA300,
+  KD_TEX_FMT_ASTC_HDR_6x6       = 0xA400,
+  KD_TEX_FMT_ASTC_HDR_8x5       = 0xA500,
+  KD_TEX_FMT_ASTC_HDR_8x6       = 0xA600,
+  KD_TEX_FMT_ASTC_HDR_8x8       = 0xA700,
+  KD_TEX_FMT_ASTC_HDR_10x5      = 0xA800,
+  KD_TEX_FMT_ASTC_HDR_10x6      = 0xA900,
+  KD_TEX_FMT_ASTC_HDR_10x8      = 0xAA00,
+  KD_TEX_FMT_ASTC_HDR_10x10     = 0xAB00,
+  KD_TEX_FMT_ASTC_HDR_12x10     = 0xAC00,
+  KD_TEX_FMT_ASTC_HDR_12x12     = 0xAD00,
+
+  KD_TEX_FMT_TYPE_MASK          = 0xF000,
+
+  KD_TEX_FMT_MASK               = 0xFFFF,
+};
+
+// Alpha handling
+enum KD_TEX_ALPHA
+{
+  KD_TEX_ALPHA_STRAIGHT         = 0x00000, // Straight (unmultiplied) alpha
+  KD_TEX_ALPHA_OPAQUE           = 0x10000, // No alpha
+  KD_TEX_ALPHA_PREMULTIPLIED    = 0x20000, // Premultiplied alpha
+
+  KD_TEX_ALPHA_MASK             = 0xF0000,
+};
+ 
+// Texture component swizzle or effect
+enum KD_TEX_SWIZ
+{
+  KD_TEX_SWIZ_RGBA              = 0x000000, // No swizzling
+  KD_TEX_SWIZ_RGB1              = 0x100000, // Normal swizzle, ignoring alpha
+  KD_TEX_SWIZ_RRR1              = 0x200000, // Luminance
+  KD_TEX_SWIZ_111R              = 0x300000, // Alpha
+  KD_TEX_SWIZ_RRRG              = 0x400000, // Luminance-Alpha
+  KD_TEX_SWIZ_RRRR              = 0x500000, // Intensity
+  KD_TEX_SWIZ_GGG1              = 0x600000, // Luminance (ETC1/BC1)
+  KD_TEX_SWIZ_111G              = 0x700000, // Alpha (ETC1/BC1)
+  KD_TEX_SWIZ_GGGA              = 0x800000, // Luminance-Alpha (BC2/BC3)
+  KD_TEX_SWIZ_GGGG              = 0x900000, // Intensity (ETC1/BC1)
+
+  KD_TEX_SWIZ_SDF               = 0xa00000, // Red channel contains a SDF
+  KD_TEX_SWIZ_RGB_SDF           = 0xb00000, // RGB8 texture with a SDF packed in the alpha channel
+  KD_TEX_SWIZ_MSDF              = 0xc00000, // Encoded MSDF in the color channels, Alpha is ignored
+
+  KD_TEX_SWIZ_MASK              = 0xF00000,
+};
+
+// Color space
+enum KD_TEX_COL
+{
+  KD_TEX_COL_REC709             = 0x0000000, // REC709/sRGB color space
+  KD_TEX_COL_REC2020            = 0x1000000, // REC2020 color space
+
+  KD_TEX_COL_MASK               = 0xF000000,
+};
+
+// Gamma
+enum KD_TEX_GAMMA
```

Comment:
Isn't transfer function a more generic term? Also used in ffmpeg. So KD_TEX_TRANSFER prefix for example.
Might as well add PQ to the list if the idea is to be able to represent images in their native format.

Reply:
Done.